### PR TITLE
pmaint eclass: new devbook format

### PR DESCRIPTION
**Pull request by arthurzam**
This is a PR for the new implementation of devbook (the format for devmanual) format for `pmaint`'s docs generator. This is intended to replace the current eclass2manpage and manpage2html usage.
You can see the current output at my [devspace](https://dev.gentoo.org/~arthurzam/devmanual/eclass-reference/index.html)
I have also tested the output of the manpage, and it looks at least the same (sometimes even better) then current once.
I have done some local edits for the devmanual makefile, so it can use this new facility (currently it uses a local hack of a symlink to gentoo repo inside src, so I still can't upstream it).

I now assume the all eclassdocs are RST formatted (except the normal special tags of course). I will traverse all eclasses and fix/update the docs, to match the new format (there are currently some last places where those are wrongly formatted). This work will be done, and sent to ML to await comments and ideas. I don't plan to merge this PR until later.

Closes: https://github.com/pkgcore/pkgcore/issues/345

### CODE REVIEW

File: src/pkgcore/ebuild/eapi.py [hunk 1]
```diff
@@ -715,3 +715,5 @@ def _combine_dicts(*mappings):
     )),
     ebd_env_options=eapi7._ebd_env_options,
 )
+
+latest_eapi = eapi8
```
Comment (ferringb) [inline src/pkgcore/ebuild/eapi.py:719]: I'm not a huge fan of 'latest' here; pkgcore's original intent was to support varying EAPI's, including developmental.  Is a developmental one 'latest' ?

TL;DR: can we call this `latest_pms_eapi` ?

Comment (ferringb) [reply src/pkgcore/ebuild/eapi.py:719]: Hell, we could have `pms_eapis = [obj1, obj2, obj3, ...]` and allow folks to just do `pms_eapis[-1]` to get latest.  Thoughts?

Comment (arthurzam) [reply src/pkgcore/ebuild/eapi.py:719]: I agree. This was a draft PR, more as a "WIP" branch for the future. You can ignore this PR from any review for the near future.

**Comment by Flowdalic** [comment]
What's the status of this? @thesamesam mentioned you said that the code is done and ist just needs the docs porting? We recently had [the case where eclass-to-manpage failed](https://github.com/gentoo/gentoo/commit/7c9bacebd7254491a72e633b523feacfec72d5ec), therefore having proper eclass verification as part of the CI would be very nice. And now the question is if we wait for this or maybe add the eclass-to-manpage awk script to the CI.
